cmd/snap/cmd_quota.go

```diff
@@ -118,13 +167,64 @@ func parseQuotas(maxMemory string) (*client.QuotaValues, error) {
 		mem = value
 	}
 
+	if cpuMax != "" {
+		countValue, percentageValue, err := parseCpuQuota(cpuMax)
+		if err != nil {
+			return nil, err
+		}
+		if percentageValue > 100 || percentageValue <= 0 {
+			return nil, fmt.Errorf("cannot use value %v: cpu quota percentage must be between 1 and 100", percentageValue)
+		}
+
+		cpuCount = countValue
+		cpuPercentage = percentageValue
+	}
+
+	if cpuSet != "" {
+		cgv, err := cgroupVersion()
+		if err != nil {
+			return nil, err
+		}
+		if cgv < 2 {
+			return nil, fmt.Errorf("cannot use CPU set with cgroup version %d", cgv)
+		}
+
+		cpuTokens := strutil.CommaSeparatedList(cpuSet)
+		for i, cpuToken := range cpuTokens {
+			cpu, err := strconv.Atoi(cpuToken)
+			if err != nil {
+				return nil, fmt.Errorf("cannot parse value for --cpu-set at position %d", i)
+			}
+			if cpu < 0 {
+				return nil, fmt.Errorf("cannot use a negative CPU number in CPU set %q", cpuSet)
+			}
```

Comment:
We could use ParseUint and have a single condition for error handling, which shows the cpuToken and doesn't mention `--cpu-set`.